otx/algorithms/detection/adapters/mmdet/task.py

```diff
@@ -0,0 +1,829 @@
+        model.eval()
+        feature_model = model.model_t if self._train_type == TrainType.SEMISUPERVISED else model
+        model = build_data_parallel(model, cfg, distributed=False)
+
+        # InferenceProgressCallback (Time Monitor enable into Infer task)
+        time_monitor = None
+        if cfg.get("custom_hooks", None):
+            time_monitor = [hook.time_monitor for hook in cfg.custom_hooks if hook.type == "OTXProgressHook"]
+            time_monitor = time_monitor[0] if time_monitor else None
+        if time_monitor is not None:
+
+            # pylint: disable=unused-argument
+            def pre_hook(module, inp):
+                time_monitor.on_test_batch_begin(None, None)
+
+            def hook(module, inp, outp):
+                time_monitor.on_test_batch_end(None, None)
+
+            model.register_forward_pre_hook(pre_hook)
+            model.register_forward_hook(hook)
+
+        explainer = explain_parameters.explainer if explain_parameters else None
+        explainer_hook = explainer_hook_selector.get(explainer.lower(), None)
+        if explainer_hook is None:
+            raise NotImplementedError(f"Explainer algorithm {explainer} not supported!")
+        logger.info(f"Explainer algorithm: {explainer}")
+
+        # Class-wise Saliency map for Single-Stage Detector, otherwise use class-ignore saliency map.
+        eval_predictions = []
+        with explainer_hook(feature_model) as saliency_hook:
+            for data in dataloader:
+                with torch.no_grad():
+                    result = model(return_loss=False, rescale=True, **data)
+                eval_predictions.extend(result)
+            saliency_maps = saliency_hook.records
+
+        # Check and unwrap ImageTilingDataset object from TaskAdaptEvalDataset
+        while hasattr(mm_dataset, "dataset") and not isinstance(mm_dataset, ImageTilingDataset):
+            mm_dataset = mm_dataset.dataset
+
+        # In the tiling case, select the first images which is map of the entire image
+        if isinstance(mm_dataset, ImageTilingDataset):
+            saliency_maps = [saliency_maps[i] for i in range(mm_dataset.num_samples)]
+
+        outputs = dict(detections=eval_predictions, saliency_maps=saliency_maps)
+
+        logger.info("run task done.")
+        detections = outputs["detections"]
+        explain_results = outputs["saliency_maps"]
+
+        self._add_explanations_to_dataset(
+            detections, explain_results, dataset, process_saliency_maps, explain_predicted_classes
+        )
+        logger.info("Explain completed")
+        return dataset
+
+    # This should be removed
+    def update_override_configurations(self, config):
+        """Update override_configs."""
+        logger.info(f"update override config with: {config}")
+        config = ConfigDict(**config)
+        self.override_configs.update(config)
+
+    # This should moved somewhere
+    def _init_deploy_cfg(self) -> Union[Config, None]:
```

Comment:
Agree that it should be moved somewhere.. I hope we can find a way to merge them in configurer or configure.